src/components/controller/Controller.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react';
+import Input from '../input/Input';
+import Output from '../output/Output';
+import History from '../history/History';
+import { hitApi } from '../../services/hit-api';
+import styles from './controllerstyles.css';
+
+class Controller extends Component {
+  state = {
+    history: [],
+    url: '',
+    method: 'GET',
+    text: ['Try searching something'],
+    body: '',
+    loading: false,
+  }
+
+  setLoading = () => this.setState({ loading: true })
+  setUnloading = () => setTimeout(() => this.setState({ loading: false }), 2000)
+
+  handleChange = ({ target }) => {
+    this.setState({ [target.name]: target.value });
+  }
+
+  updateHistory = newHistoryItem => {
+    const newHistoryItemData = newHistoryItem;
+    const now = new Date().toString();
+    const newHistory = this.state.history;
+
+    newHistoryItemData.date = now;
+    newHistory.push(newHistoryItemData);
+    this.setState({ history: newHistory }); 
+  }
```

Comment:
This looks good but it's a nice opportunity to use the spread operator to update the array in state